records/labels/spec.go

```diff
@@ -0,0 +1,58 @@
+package labels
+
+import (
+	"fmt"
+)
+
+const (
+	// SeparatorChar delimits concatenated labels.
+	SeparatorChar = '.'
+
+	DNS952MaxLength  int = 24
+	DNS1123MaxLength int = 63
+
+	// see http://www.rfc-base.org/txt/rfc-952.txt
+	HostNameSpecType952 HostNameSpecType = iota
+
+	// see http://www.rfc-base.org/txt/rfc-1123.txt
+	HostNameSpecType1123
```

Comment:
Yep, probably. I'll keep the constants for length though - trying to avoid
introducing magic numbers not assigned to constants.

On Thu, Jun 25, 2015 at 10:14 PM, Tomás Senart <notifications@github.com>
wrote:

> In records/labels/spec.go
> <https://github.com/mesosphere/mesos-dns/pull/177#discussion_r33324053>:
>
> > +import (
> > +	"fmt"
> > +)
> > +
> > +const (
> > +	// SeparatorChar delimits concatenated labels.
> > +	SeparatorChar = '.'
> > +
> > +	DNS952MaxLength  int = 24
> > +	DNS1123MaxLength int = 63
> > +
> > +	// see http://www.rfc-base.org/txt/rfc-952.txt
> > +	HostNameSpecType952 HostNameSpecType = iota
> > +
> > +	// see http://www.rfc-base.org/txt/rfc-1123.txt
> > +	HostNameSpecType1123
>
> I believe this can be further simplified. No need for HostNameSpecType.
>
> var (
>     RFC952   = HostNameSpec{table:  makeDNS952Table(), maxlen: 24}
>     RFC1123  = HostNameSpec{table:  makeDNS1123Table(), maxlen: 63}
> )
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesosphere/mesos-dns/pull/177/files#r33324053>.
>